astropy/coordinates/funcs.py

```diff
@@ -279,3 +254,120 @@ def get_constellation(coord, short_name=False, constellation_list='iau'):
+        The objects to concatenate
+
+    Returns
+    -------
+    rep : `~astropy.coordinates.BaseRepresentation` subclass
+        A single representation object with its data set to the concatenation of
+        all the elements of the input sequence of representations.
+    """
+    if not isinstance(reps, (Sequence, np.ndarray)):
+        raise TypeError('Input must be a list or iterable of representation '
+                        'objects.')
+
+    # First, validate that the represenations are the same, and
+    # concatenate all of the positional data:
+    rep_type = type(reps[0])
+    if any(type(r) != rep_type for r in reps):
+        raise TypeError('Input representations must all have the same type.')
+
+    # Construct the new representation with the concatenated data from the
+    # representations passed in
+    values = _concatenate_components(reps,
+                                     rep_type.attr_classes.keys())
+    new_rep = rep_type(*values)
+
+    has_diff = any('s' in rep.differentials for rep in reps)
+    if has_diff and any('s' not in rep.differentials for rep in reps):
+        raise ValueError('Input representations must either all contain '
+                         'differentials, or not contain differentials.')
+
+    if has_diff:
+        dif_type = type(reps[0].differentials['s'])
+
+        if any('s' not in r.differentials or
+                type(r.differentials['s']) != dif_type
+               for r in reps):
+            raise TypeError('All input representations must have the same '
+                            'differential type.')
+
+        values = _concatenate_components([r.differentials['s'] for r in reps],
+                                         dif_type.attr_classes.keys())
+        new_dif = dif_type(*values)
+        new_rep = new_rep.with_differentials({'s': new_dif})
+
+    return new_rep
+
+
+def concatenate(coords):
+    """
+    Combine multiple coordinate objects into a single
+    `~astropy.coordinates.SkyCoord`.
+
+    "Coordinate objects" here mean frame objects with data,
+    `~astropy.coordinates.SkyCoord`, or representation objects.  Currently,
+    they must all be in the same frame, but in a future version this may be
+    relaxed to allow inhomogenous sequences of objects.
+
+    Parameters
+    ----------
+    coords : sequence of coordinate objects
+        The objects to concatenate
+
+    Returns
+    -------
+    cskycoord : SkyCoord
+        A single sky coordinate with its data set to the concatenation of all
+        the elements in ``coords``
+    """
+    if getattr(coords, 'isscalar', False) or not isiterable(coords):
+        raise TypeError('The argument to concatenate must be iterable')
+
+    scs = [SkyCoord(coord) for coord in coords]
+
+    # Check that all frames are equivalent
+    for sc in scs[1:]:
+        if not sc.is_equivalent_frame(scs[0]):
+            raise ValueError("All inputs must have equivalent frames: "
```

Comment:
with a trivial `transform_to`, we could now make this put them all in the same frame... maybe that's a good enhancement (if you don't think it's a good idea it shouldn't block this PR, though)

Reply:
My gut reaction is to make the user decide, rather than to assume that the frame of the 1st object is special. So, something to consider for future but I'd rather not do that now.